includes/admin/admin-filters.php

```diff
@@ -26,7 +27,56 @@
  *
  * @return  mixed
  */
-function _give_sanitize_number_decimals_setting_field( $value ) {
+function __give_sanitize_number_decimals_setting_field( $value ) {
+	$value_changed = false;
+	$old_value     = $value;
+
+	if ( isset( $_POST['decimal_separator'] ) ) {
+		$value         = ! empty( $_POST['decimal_separator'] ) ? $value : 0;
+		$value_changed = true;
+	}
+
+	// Check if
+	if ( $value_changed && ( $old_value != $value ) ) {
+		Give_Admin_Settings::add_error( 'give-number-decimal', __( '\'Number of Decimals\' automatically set to zero because \'Decimal Separator\' can not empty.', 'give' ) );
+	}
+
+	$value = absint( $value );
+
+	if( 6 <= $value ) {
+		$value = 5;
```

Comment:
@DevinWalker Let me know you want to show admin message while we auto set number of decimal to 5 when admin trying to set it to 6 or more than 6